python/llm/src/bigdl/llm/optimize.py

```diff
@@ -208,6 +208,8 @@ def optimize_model(model, low_bit='sym_int4', optimize_llm=True, modules_to_not_
         to `True` when running BigDL-LLM on GPU on Windows. Default to be `False`.
     :param lightweight_bmm: Whether to replace the torch.bmm ops, may need to set it
         to `True` when running BigDL-LLM on GPU on Windows. Default to be `False`.
+    :param quantize_kv_cache: Whether to quantize kv cache, it can be only used on GPU.
+        Default to be `False`.
```

Comment:
Instead of having the user to manually specify it, I think we can automatically control its behavior as follows:
```python
def quantize_kv_cache:
    if ENV BIGDL_QUANTIZE_KV is defined
       return BIGDL_QUANTIZE_KV 
    elif it's running on CPU or qtype is FP16/BF16
       return fasle
    elif
       return true

def qwen_attention_forward:
    if quantize_kv_cache()
       qwen_quantized_attention_forward()
    else
       qwen_unquantized_attention_forward()
```

Reply:
a small question, if user use `from_pretrained` API, then this convert will happen in a very early stage, at this time, we don't know whether user will put this model to XPU and whether user will convert it to other dtype

Reply:
so we can only check these condition inside the `qwen_attention_forward` function, shall we do that?

Reply:
> so we can only check these condition inside the `qwen_attention_forward` function, shall we do that?

Yes, we need to check inside `qwen_attention_forward`